### DISCUSSION TITLE
Dreame L10 not accessible

### DISCUSSION THREAD

**Discussion by tdcockers** [Hypfer/Valetudo#1995] [Q&A / Support]
Similar to #1780 (probably happened about the same time, but I haven't been able to circle back to it until now) my robot suddenly and without any intervention that I'm aware of stopped responding to requests over the network (MQTT, web interface, companion app).

I can access over serial, and after resetting wifi I can access over ssh via IP if I had the necessary ssh key but don't know the root password, so I think everything is fine on the hardware side but I can't figure out what my next step is to get it talking again?

Thanks

**Comment by tdcockers** [comment]
Actually, I mis-typed a command, and it gave me a possible lead that 'file system is not writeable'. Ran df -h, /dev/root is 100% full. Is there something I can or should clean up in there?

**Comment by stefanmd023** [comment]
If you can't access the valetudo webapp to setup the Wifi, then valetudo was removed. If you don't have your key to ssh in and put valetudo manually, and you're unable to get the password right, then you will need to dustbuild and flash over UART again. Follow instructions for your robot from the guide.

edit: the statement "your robot factory reset" was incorrect

**Reply by Hypfer** [reply]
>  then your robot factory reset and valetudo was removed

It's _important_ to point out that that isn't correct. The robot didn't factory-reset.
It likely suffered from filesystem corruption of the ext4 filesystem of the /data partition, leading to the firmware creating a new filesystem on there and with that it losing the config, the map, the wifi config and the Valetudo binary.

A factory-reset would be an intentional process with logic intended to reset the system.
This is just data corruption.

I'm being very specific here because "random factory resets" was an issue on the roborock v1 and s5 once and there, that was an actual factory reset that was triggered by a firmware watchdog noticing the high ram usage caused by Valetudo running alongside the firmware.

As people search these replies for answers, they _will_ run with what they read here and that's how rumors spread. I have 0 interest in reading something along the lines of "random factory resets are back!!111" on reddit.
When in doubt, **please** don't answer at all instead of answering only 70% correct. It saves me a lot of headache later on. If it is a question that requires answering, I will answer eventually. Please give me some time to do so.




Anyway, this corruption _seems_ to be unrelated to Valetudo as Valetudo doesn't do any writes and it can also happen to partitions mounted read-only like those containing the calibration data.
It also seems to happen with robots running stock firmware.

Because of that, hold on to your backups. They might save your robot if that ever happens to it.



As the rootfs is unaffected by this, there's still an sshd running to which you can connect either with your ssh key or - if you didn't disable it - via password login. The password can be calculated from the SN as documented in the rooting instructions.

Then, just flash a fresh dustbuilder firmware with valetudo prepackaged via ssh

**Reply by tdcockers** [reply]
Perfect thanks. I had forgotten the part about calculating root password, have re-flashed and all is back up and running now :)

**Reply by morremeyer** [reply]
@Hypfer Thanks for this detailed explanation. I just stumbled into the situation that my L10 Pro "dropped" from the network. I investigated, and after SSHing into it it looked like a freshly rooted robot.

I then followed the steps in https://valetudo.cloud/pages/installation/dreame.html starting with **Phase 3: Install Valetudo** to re-install Valetudo. This worked perfectly fine, and it's now running again (minus the data loss, but I don't mind that).

Did I understand your description correctly that this issue of "Valetudo vanishing from the robot" is something that we don't have any influence over because if the ext4 corruption happens, it just happens?

**Reply by Hypfer** [reply]
Yes, that is what I wrote
